nixos/modules/image/interactive.nix

```diff
@@ -0,0 +1,95 @@
+# An interactive image built using systemd-repart.
+#
+# This is a higher-level abstraction built on top of repart.nix but designed
+# to give you an interactive (i.e. with Nix available and working image.) image.
+#
+# You can use this by importing this module in your config.
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  rootPartitionLabel = "root";
+
+  closureInfo = pkgs.closureInfo {
+    rootPaths = [ config.system.build.toplevel ];
+  };
+
+  # Build the nix state at /nix/var/nix for the image
+  #
+  # This does two things:
+  # (1) Setup the initial profile
+  # (2) Create an initial Nix DB so that the nix tools work
+  nixState = pkgs.runCommand "nix-state" { nativeBuildInputs = [ pkgs.buildPackages.nix ]; } ''
```

Comment:
Do we need the `buildPackages` ? I thought `nativeBuildInputs` rewrites this for us